WordPress/openverse-frontend
# Support for video page and additional content types

## Fixes


Fixes #631 by @panchovm 

## Description


This PR adds support for the video page with Additional sources form. It also refactors the way search types are handled to enable using a non-supported search type, too.

I am setting this as medium priority, but it may be 'high' because the `model_3d` work depends on these changes.

I have a question about what to do when a user tries to search from a non-supported type search page (i.e., video). When the user enters a search term on the Video page and clicks 'Search', we can:
1. automatically search for All Content - redirect to the All Content page
2. update the path and set the search term in the store, but do nothing else. This way, the user can change the search type using the content switcher, and be redirected to the same query for other search types? Here, we can also use some 'popup'/ 'toast' indication saying that search for video is not available.
3. do nothing (maybe with some kind of indication like a toast saying that search is not available for 'video')

I think I would expect the second option if I was the user. What do you think, @panchovm ?

## Testing Instructions

Open the video page (http://localhost:8443/search/video or http://localhost:8443/search/video/?q=cat). It should not be broken, and should show the Additional sources buttons that redirect to video search on the provider sites.
Changing the search type in the content switcher should also work.

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
